Site_Files/src/main/webapp/script.js

```diff
@@ -73,3 +59,92 @@ function search(){
     });
   });
 }
+
+/**
+ * Retrieves the posts from the /inf servlet and injects it into the page
+ */
+function getAllPosts() {
+  fetch('/inf').then(response => response.json()).then((posts) => {
+    const postsListElement = document.getElementById('posts-container');
+
+    // for each message, display it in a list
+    posts.forEach((post) => {
+        postsListElement.appendChild(createPostElement(post));
+        loadStarRatings("#rating-" + post.id);
+        storeStarRatings("#rating-" + post.id);
+    });
+  });
+}
+
+/**
+ * Create each post element using the DataStore information
+ */
+function createPostElement(postData) {
+  const liElement = document.createElement('li');
+  liElement.className = "post";
+  liElement.id = "post-" + postData.id;
+
+  const nameElement = document.createElement('span');
+  nameElement.innerText = postData.name + " (" + postData.email + ")" ;
+
+  const typeElement = document.createElement('span');
+  typeElement.innerText = postData.type;
+
+  const dateElement = document.createElement('span');
+  const date = new Date(postData.timestamp);
+  const dateTime = " - " + date.toDateString() + " " + date.getHours() + ":" + date.getMinutes();
+  dateElement.innerText = dateTime;
+
+  const commentElement = document.createElement('p');
+  commentElement.innerText = postData.comment;
+  
+  const starRatingElement = document.createElement('input');
+  starRatingElement.className = 'rating-loading';
+  starRatingElement.type = 'text';
+  starRatingElement.id = 'rating-' + postData.id;
+  starRatingElement.value = postData.rating;
+  
+  liElement.appendChild(nameElement);
+  liElement.appendChild(typeElement);
+  liElement.appendChild(dateElement);
+  liElement.appendChild(commentElement);
+  liElement.appendChild(starRatingElement);
+  
+  return liElement;
+}
+
+/**
+ * Loads the star rating settings into the page.
+ */
+function loadStarRatings(id){
+    $(document).ready(function(){
+        $(id).rating({
+        step: 1,
+        starCaptions: {1: 'Very Poor', 2: 'Poor', 3: 'Ok', 4: 'Good', 5: 'Very Good'},
+        starCaptionClasses: {1: 'text-danger', 2: 'text-warning', 3: 'text-info', 4: 'text-primary', 5: 'text-success'}
+        });
+    });
+}
+
+/**
+ * Stores the star rating to the DataStore.
+ */
+function storeStarRatings(id){
```

Comment:
I would consider renaming this function to something else. "storeStarRatings" implies that you're storing the star ratings when this function is called, when in reality this function just sets up the handler to do so.